server/src/services/workflow.service.ts

```diff
@@ -56,7 +56,7 @@ export class WorkflowService extends BaseService {
     }
 
     const workflow = await this.findOrFail(id);
-    const trigger = this.getTriggerOrFail(workflow.triggerType);
+    const trigger = this.getTriggerOrFail(dto.triggerType ?? workflow.triggerType);
```

Comment:
Probably should just call this `const contextType = getContextForTrigger(...)`;`